assets/terraform/gce/bootstrap/suse.sh

```diff
@@ -78,6 +78,26 @@ if ! grep -qs "$etcd_dir" /proc/mounts; then
   mount $etcd_dir
 fi
 
+# grow-root-fs expands '/' to use all available space on the device.
+#
+# The GCP sles-12-sp5-v20200610 image doesn't recognize extra space in drives
+# by default, and limits the '/' partition to 10G without this kick.
+#
+# See https://cloud.google.com/compute/docs/disks/add-persistent-disk#resize_partitions
+function grow-root-fs {
+  echo "Filesystem utilization before grow-root-fs:"
+  df -h
+  local root_fs_dev=$(egrep '^(.*) / xfs' /proc/mounts | awk '{print $1}')
+  local partion_number=$(echo $root_fs_dev | egrep -o '([0-9]+)$')
```

Comment:
> My only guess so far on mapping the device path to partition number for growpart is maybe subtracting partition's MINOR from disk's MINOR from /proc/partitions - not sure if I'm blowing things out of proportion here.

Not at all.  I also thought about this -- but I thought it'd be better to put the code up for review & tap the collective wisdom instead of fiddling further with the dozen different options for looking at device/partion info.  I'll try this out to if it seems cleaner & more future proof.

> If I'm not mistaken - this sequence will also fail if the mount point is not a partition - do you need to handle this case (i.e. parse it as empty)?

The suse-cloud/sles-12-sp5-v20200227 image I've tested thus far has the following partition scheme:

```
robotest@walt-suse12-node-0:~> lsblk                                                                                                                                                          
NAME   MAJ:MIN RM SIZE RO TYPE MOUNTPOINT                                                                                                                                                     
sda      8:0    0  64G  0 disk                                                                                                                                                                
├─sda1   8:1    0   2M  0 part                                                                                                                                                                
├─sda2   8:2    0  20M  0 part /boot/efi                                                                                                                                                      
└─sda3   8:3    0  10G  0 part /                                                                                                                                                              
sdb      8:16   0  50G  0 disk /var/lib/gravity/planet/etcd
```

I definitely can add in smarter logic (e.g. it also makes me a bit uncomfortable this code assumes xfs), but I'd like to be able to test any code paths I put in.  I plan to test against:

- [x] suse-cloud/sles-12-sp5-v20200227 (currently in robotest)
- [ ] suse-cloud/sles-12-sp5-v20200610 (latest image)
- [ ] suse-cloud/sles-15-sp1-v20200415 (currently in robotest)
- [ ] suse-cloud/sles-15-sp1-v20200610 (latest image)

If none of these turn up issues with the current partition dependant strategy, are you ok taking it without further hardening?  I don't want to create custom images to test a hypothetical case, that seem low ROI. And committing untested future proofing strikes me as less useful than ignoring the non-partitioned case and fixing it if/when it becomes a problem.

Reply:
I refactored this into something a bit more robust that uses lsblk and `/sys/class/block` in 
c91d344.  It still depends on the root fs being a partition.

I haven't done another end-to-end test yet but I did a one-off run by cutting & pasting onto a SUSE vm, and will get another end-to-end test before merging.

```
+ grow-root-fs
+ echo 'Filesystem utilization before grow-root-fs:'
Filesystem utilization before grow-root-fs:
+ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs        3.9G     0  3.9G   0% /dev
tmpfs           3.9G     0  3.9G   0% /dev/shm
tmpfs           3.9G   18M  3.9G   1% /run
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/sda3        64G  1.6G   63G   3% /
/dev/sda2        20M  3.4M   17M  17% /boot/efi
tmpfs           797M     0  797M   0% /run/user/1003
/dev/sdb         49G   53M   47G   1% /var/lib/gravity/planet/etcd
++ findmnt --noheadings -o SOURCE /
+ local root_fs_dev=/dev/sda3
++ lsblk --noheadings -o kname /dev/sda3
+ local root_fs_dev_name=sda3
++ cat /sys/class/block/sda3/partition
+ local partition_number=3
++ lsblk --noheadings -o pkname /dev/sda3
+ local parent_dev_name=sda
+ local parent_dev=/dev/sda
+ growpart /dev/sda 3
NOCHANGE: partition 3 is size 134170591. it cannot be grown
+ xfs_growfs /
meta-data=/dev/sda3              isize=512    agcount=26, agsize=653887 blks
         =                       sectsz=512   attr=2, projid32bit=1
         =                       crc=1        finobt=0 spinodes=0 rmapbt=0
         =                       reflink=0
data     =                       bsize=4096   blocks=16771323, imaxpct=25
         =                       sunit=0      swidth=0 blks
naming   =version 2              bsize=4096   ascii-ci=0 ftype=1
log      =internal               bsize=4096   blocks=2560, version=2
         =                       sectsz=512   sunit=0 blks, lazy-count=1
realtime =none                   extsz=4096   blocks=0, rtextents=0
+ echo 'Filesystem utilization after grow-root-fs:'
Filesystem utilization after grow-root-fs:
+ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs        3.9G     0  3.9G   0% /dev
tmpfs           3.9G     0  3.9G   0% /dev/shm
tmpfs           3.9G   18M  3.9G   1% /run
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/sda3        64G  1.6G   63G   3% /
/dev/sda2        20M  3.4M   17M  17% /boot/efi
tmpfs           797M     0  797M   0% /run/user/1003
/dev/sdb         49G   53M   47G   1% /var/lib/gravity/planet/etcd
```